### ISSUE TITLE
Stopped working

### ISSUE THREAD

**Issue by karelbilek** [zertosh/eslint-plugin-flow-vars#5]
This plugin stopped working with newest eslint and babel-eslint.

This is what I got:

```
Cannot read property 'type' of undefined
TypeError: Cannot read property 'type' of undefined
    at isFunction (/usr/local/lib/node_modules/eslint/lib/rules/no-use-before-define.js:49:28)
    at isFunctionOrOuterClass (/usr/local/lib/node_modules/eslint/lib/rules/no-use-before-define.js:74:12)
    at /usr/local/lib/node_modules/eslint/lib/rules/no-use-before-define.js:162:17
    at Array.forEach (native)
    at findVariablesInScope (/usr/local/lib/node_modules/eslint/lib/rules/no-use-before-define.js:149:26)
    at EventEmitter.findVariables (/usr/local/lib/node_modules/eslint/lib/rules/no-use-before-define.js:184:9)
    at emitOne (events.js:83:20)
    at EventEmitter.emit (events.js:170:7)
    at NodeEventGenerator.leaveNode (/usr/local/lib/node_modules/eslint/lib/util/node-event-generator.js:51:22)
    at CodePathAnalyzer.leaveNode (/usr/local/lib/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:611:23)
```

If I either stop using `no-use-before-define` or the `flow-vars` rule, this error stop appearing; however, if I remove the `flow-vars` rules, I get all the annoying "errors" from eslint that this module is for.

**Comment by karelbilek** [comment]
I think I have been able to isolate where this happens.

It seems to happen whenever you use a type or class that is defined in file in `[libs]` part of flowconfig, which means the type/class is "global" (and you don't need to import it specifically to use it with Flow).

Once I rewrote that so the types don't flow as "globals" and they are declared in a module, and they are explicitly imported, this error stopped appearing.

**Comment by zertosh** [comment]
I can't repro using `babel-eslint@6.0.0` and `eslint@2.5.1`. I tried your original `chrome.js` with

```
{
  "parser": "babel-eslint",
  "plugins": [
    "flow-vars"
  ],
  "rules": {
    "no-undef": 1,
    "no-unused-vars": 1,
    "no-use-before-define": 1,
    "flow-vars/define-flow-type": 1,
    "flow-vars/use-flow-type": 1
  }
}
```

**Comment by karelbilek** [comment]
Interesting. I will try to create some small usecase

**Comment by maggie-wang-bbt** [comment]
any update on this one?

**Comment by AriaFallah** [comment]
> It seems to happen whenever you use a type or class that is defined in file in [libs] part of flowconfig, which means the type/class is "global" (and you don't need to import it specifically to use it with Flow

I have the same problem here.

**Comment by zertosh** [comment]
Can someone give me a file and a config so I can repro?

**Comment by AriaFallah** [comment]
@zertosh 

Here's a small example I made inside my React project

.flowconfig

```
[ignore]
.*/node_modules/fbjs/lib/*

[include]

[libs]
types.js

[options]
module.system=node
suppress_comment= \\(.\\|\n\\)*\\$FlowIgnore
```

types.js

``` javascript
// @flow

type Pos = { x: number, y: number }
```

test.js

``` javascript
// @flow

function logPos(c: Pos) {}
```

.babelrc

``` json
{
  "presets": ["react", "es2015", "stage-1"]
}
```

.eslintrc

``` json
{
  "extends": ["airbnb"],
  "parser": "babel-eslint",
  "rules": {
    "comma-dangle": 0,
    "no-console": 0,
    "semi": [2, "never"],
    "space-before-function-paren": 0,
    "no-multi-spaces": 0,
    "no-param-reassign": 0,
    "no-return-assign": 0,
    "no-confusing-arrow": [2, { "allowParens": true }],
    "no-use-before-define": [2, "nofunc"],
    "flow-vars/define-flow-type": 1,
    "flow-vars/use-flow-type": 1
  },
  "plugins":["react", "flow-vars"]
}
```

and the error:

```
Error: Cannot read property 'type' of undefined
TypeError: Cannot read property 'type' of undefined
    at isFunction (/Users/Developer/workspace/projects/games/reactive-snake/node_modules/eslint/lib/rules/no-use-before-define.js:49:28)
    at /Users/Developer/workspace/projects/games/reactive-snake/node_modules/eslint/lib/rules/no-use-before-define.js:162:17
    at Array.forEach (native)
    at findVariablesInScope (/Users/Developer/workspace/projects/games/reactive-snake/node_modules/eslint/lib/rules/no-use-before-define.js:149:26)
    at EventEmitter.ruleDefinition.Program:exit (/Users/Developer/workspace/projects/games/reactive-snake/node_modules/eslint/lib/rules/no-use-before-define.js:196:17)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.leaveNode (/Users/Developer/workspace/projects/games/reactive-snake/node_modules/eslint/lib/util/node-event-generator.js:51:22)
    at CodePathAnalyzer.leaveNode (/Users/Developer/workspace/projects/games/reactive-snake/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:628:23)
    at CommentEventGenerator.leaveNode (/Users/Developer/workspace/projects/games/reactive-snake/node_modules/eslint/lib/util/comment-event-generator.js:111:23)
```

**Comment by zertosh** [comment]
Thanks @AriaFallah, that was really helpful! I published the fix as `eslint-plugin-flow-vars@0.3.0`.

**Comment by karelbilek** [comment]
Thanks a lot! 🍺
